SerialPrograms/Source/PokemonLA/Inference/PokemonLA_ShinySoundDetector.cpp

```diff
@@ -0,0 +1,78 @@
+/*  Shiny Sound Detector
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "CommonFramework/Tools/AudioFeed.h"
+#include "PokemonLA_ShinySoundDetector.h"
+
+#include <QString>
+
+#include <cfloat>
+#include <iostream>
+using std::cout;
+using std::endl;
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonLA{
+
+
+
+ShinySoundDetector::ShinySoundDetector(Logger& logger)
+    : AudioInferenceCallback("ShinySoundDetector")
+    , m_logger(logger)
+{
+    // TODO : handle case with sample rate mismatch between file and capture card:
```

Comment:
What happens if there is a match right now? Can you make it throw an error for now? We'll come back and add it later, but I don't want it to silently fail as is.